### DISCUSSION TITLE
How make database class for non-crud apllications?

### DISCUSSION THREAD

**Discussion by loss-and-quick** [mikro-orm/mikro-orm#4361] [Q&A] [answered]
I want to do something like this:
``` ts
class Database {
    async createEntity(dto: EntityDTO) {
        return entity;
    }
    async getEntities() {
        return entities;
    }
}
```

But MicroORM looks like it is only for CRUD applications

**Comment by B4nan** [comment]
I have no idea what you mean by that.

**Reply by loss-and-quick** [reply]
I mean that I want to make a Database class that has an aliases for convenient work with the database via MicroORM
I'm asking this question because I don't understand how i can properly inherit the MicroORM class so that i can do something like this

**Reply by B4nan** [reply]
You should not inherit it 🤷 Still no idea what you are up to, you don't need any wrappers around the ORM class, you should work with the EntityManager. This is not a query builder, you don't have to create your own abstractions around it.

https://mikro-orm.io/docs/entity-manager

Here are some more examples:

https://stackblitz.com/edit/mikro-orm-getting-started-guide-cp-1?file=src%2Fserver.ts
https://stackblitz.com/edit/mikro-orm-getting-started-guide-cp-2?file=src%2Fserver.ts

**Reply by loss-and-quick** [reply]
> You should not inherit it 🤷 Still no idea what you are up to, you don't need any wrappers around the ORM class, you should work with the EntityManager. This is not a query builder, you don't have to create your own abstractions around it.
> 
> https://mikro-orm.io/docs/entity-manager
> 
> Here are some more examples:
> 
> https://stackblitz.com/edit/mikro-orm-getting-started-guide-cp-1?file=src%2Fserver.ts https://stackblitz.com/edit/mikro-orm-getting-started-guide-cp-2?file=src%2Fserver.ts

Okay,lets change question: I don't want to drag MicroORM all over the project. my project( which is desktop application) 
How i can do this?
I want to create some wrapper of MicroORM with aliases of commands

**Comment by loss-and-quick** [comment, marked as answer]
Great idea is to use dependency injection
